Dockerfile

```diff
@@ -20,11 +20,15 @@ ENV WORLDPATH=/world
 ENV CONFIGPATH=/world
 ENV LOGPATH=/tshock/logs
 
+# add terraria user to run as
+RUN groupadd -r terraria && \
+    useradd -m -r -g terraria terraria
```

Comment:
Unfortunately, this `RUN` will have to exist as its own layer due to the `COPY` layers that follow. It would be possible to do this with the rest of the `RUN`s, but that would result in needing to `chown /tshock` which would result in duplicated files in the layering that increase the size of the container.

Reply:
I take it back, this was able to be done if it's okay to move the `COPY`s.